awx_collection/plugins/modules/tower_job_template.py

```diff
@@ -12,410 +13,430 @@
+      required: False
+      type: str
+      default: ''
+    host_config_key:
+      required: False
+      type: str
+      default: ''
+    job_slice_count:
       description:
-        - Prompt user for inventory on launch.
+        - The number of jobs to slice into at runtime. Will cause the Job Template to launch a workflow if value is greater than 1.
+      required: False
+      type: int
+      default: '1'
+    ask_skip_tags_on_launch:
+      required: False
       type: bool
-      default: 'no'
-    ask_credential:
+      default: 'False'
+    name:
       description:
-        - Prompt user for credential on launch.
-      type: bool
-      default: 'no'
-    survey_enabled:
+        - Name of this job template.
+      required: True
+      type: str
+    new_name:
       description:
-        - Enable a survey on the job template.
-      version_added: 2.7
-      type: bool
-      default: 'no'
-    survey_spec:
+        - To use when changing a team's name.
+      required: True
+      type: str
+    use_fact_cache:
       description:
-        - JSON/YAML dict formatted survey definition.
-      version_added: 2.8
+        - If enabled, Tower will act as an Ansible Fact Cache Plugin; persisting facts at the end of a playbook run to the database and caching facts for use by Ansible.
+      required: False
+      type: bool
+      default: 'False'
+    extra_vars:
+      required: False
       type: dict
+      default: ''
+    verbosity:
+      required: False
+      type: str
+      default: '0'
+      choices:
+        - '0'
+        - '1'
+        - '2'
+        - '3'
+        - '4'
+        - '5'
+    ask_credential_on_launch:
       required: False
-    become_enabled:
-      description:
-        - Activate privilege escalation.
       type: bool
-      default: 'no'
-    concurrent_jobs_enabled:
+      default: 'False'
+    project:
+      required: False
+      type: str
+    webhook_service:
       description:
-        - Allow simultaneous runs of the job template.
-      version_added: 2.7
-      type: bool
-      default: 'no'
+        - Service that webhook requests will be accepted from
+      required: False
+      type: str
+      choices:
+        - 'github'
+        - 'gitlab'
     timeout:
       description:
-        - Maximum time in seconds to wait for a job to finish (server-side).
+        - The amount of time (in seconds) to run before the task is canceled.
+      required: False
       type: int
-    custom_virtualenv:
-      version_added: "2.9"
-      description:
-        - Local absolute file path containing a custom Python virtualenv to use.
-      type: str
+      default: '0'
+    ask_scm_branch_on_launch:
       required: False
-      default: ''
+      type: bool
+      default: 'False'
     state:
       description:
         - Desired state of the resource.
-      default: "present"
       choices: ["present", "absent"]
+      default: "present"
       type: str
```

Comment:
The git diff for this DOCUMENTATION is pretty unreadable. What did you do that led to these changes? Did they get re-ordered? Did you run the generator tool? We shouldn't rename fields without a deprecated alias put in place. Renaming `ask_credential` to `ask_credential_on_launch` is not so simple.

Reply:
Yes, this was run from the auto-generator. I recently realized that the POST section of the OPTIONS page is a hash and not an array so I think we will need to do a sort on all of the fields so that the generator spits the fields out in a consistent manner.

As for the aliases, I think they should work. In the case of ask credentials on launch we have this documentation:
```
    ask_credential_on_launch:
      description:
        - NO DESCRIPTION GIVEN IN THE TOWER API
      required: False
      type: bool
      default: 'False'
      aliases:
        - ask_credential
 ```

And in the args_spec its:
```
        ask_credential_on_launch=dict(required=False, type='bool', default='False', aliases=['ask_credential']),
```

So even though the field is renamed the old names should remain functional for portability.